ui/src/app/shared/components/object-editor/object-editor.tsx

```diff
@@ -91,7 +91,12 @@ export const ObjectEditor = <T extends any>({type, value, buttons, onChange}: Pr
                         renderIndentGuides: false,
                         scrollBeyondLastLine: true
                     }}
-                    onChange={() => setIsModified(true)}
+                    onChange={v => {
+                        if (!isModified) {
+                            setText(v);
+                            setIsModified(true);
+                        }
+                    }}
```

Comment:
Fixes https://github.com/argoproj/argo-workflows/issues/5493

Reply:
I have a fix in another PR for this, I think it needs to be a bit more involved. Can you revert this one change please?

Reply:
Absolutely